torchvision/prototype/datasets/utils/_dataset.py

```diff
@@ -62,50 +87,31 @@ def __init__(
         self.homepage = homepage
         self.license = license
 
-        self._valid_options = valid_options or dict()
-        self._configs = tuple(
-            DatasetConfig(**dict(zip(self._valid_options.keys(), combination)))
-            for combination in itertools.product(*self._valid_options.values())
-        )
-
-        self.extra = FrozenBunch(extra or dict())
-
-    @property
-    def default_config(self) -> DatasetConfig:
-        return self._configs[0]
-
-    @staticmethod
-    def read_categories_file(path: pathlib.Path) -> List[List[str]]:
+    def read_categories_file(self, path: pathlib.Path) -> List[List[str]]:
         with open(path, newline="") as file:
             return [row for row in csv.reader(file)]
 
     def make_config(self, **options: Any) -> DatasetConfig:
```

Comment:
What was the original reason for doing the config creation / validation separately from `__init__()`?

Reply:
I've explained in https://github.com/pytorch/vision/pull/5369#discussion_r798679279. Let's keep the discussion there.